R/model_run.R

```diff
@@ -28,6 +29,18 @@ model_run <- function(parameters, model_version) {
                                 "hospitalised",
                                 "dead")]
 
+  # group results data by vaccination group rather than compartment
+  # to return total vaccinations
+  vax_time_series <- dplyr::group_by(
+    model_results$model_data, time, vaccine_group
+  )
+  vax_time_series <- dplyr::summarise(vax_time_series, value = sum(value))
+  vax_time_series <- tidyr::pivot_wider(
+    vax_time_series,
+    id_cols = "time", values_from = "value", names_from = "vaccine_group"
+  )
```

Comment:
Just a few changes to suggest:

1. Accessing data using `daedalus::get_data()` - that allows me to update the output structure `model_results` without breaking this operation.
2. Filtering and summarising rather than summarising and pivoting - no great preference there.

```suggestion
model_data <- daedalus::get_data(model_results)
vax_time_series <- dplyr::filter(
  model_data,
  vaccine_group == "vaccinated"
)
vax_time_series <- dplyr::summarise(
  vax_time_series,
  vaccinated = sum(value), .by = "time"
)
```

Reply:
Thanks! That does look neater. I copied the original pattern from the time_series code you put together. Shall I update that too?

Reply:
So in the other time series you're going after three different variables, one per column, so you need the pivot to wide format. Here it's only the one for now _but_ if we need to show the daily vaccinations as well, then you'd need to filter as `vaccine_group %in% c("vaccinated", "new_vaccinations")` and then pivot wider.

Reply:
Alternatively, you could run `get_new_vaccinations(model_results$model_data)` and it'll give you a data.frame of time and new vaccinations, and you can add the new vaccinations to the `time_series` list.

Reply:
Thanks! Yes, I was expecting to do the latter, so will just make your original suggested change.

Reply:
Hm, getting this error from `get_data`, any ideas?
```
Error in `UseMethod("get_data")`: no applicable method for 'get_data' applied to an object of class "list"
```

Reply:
Ah, sorry that's a mocked test failure, it works fine manually.